languagetool-server/src/main/java/org/languagetool/server/V2TextChecker.java

```diff
@@ -39,9 +45,40 @@ class V2TextChecker extends TextChecker {
 
   private static final String JSON_CONTENT_TYPE = "application/json";
   private static final Pattern COMMA_WHITESPACE_PATTERN = Pattern.compile(",\\s*");
+  private static final Logger logger = LoggerFactory.getLogger(V2TextChecker.class);
+
+  private static final Map<String,Float> ruleIdToConfidence = new HashMap<>();
 
   V2TextChecker(HTTPServerConfig config, boolean internalServer, Queue<Runnable> workQueue, RequestCounter reqCounter) {
     super(config, internalServer, workQueue, reqCounter);
+    try {
+      loadConfidenceMap();
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private void loadConfidenceMap() throws IOException {
+    if (config.getRuleIdToConfidenceFile() != null) {
+      logger.info("Loading confidence map for rules from " + ruleIdToConfidence);
+      List<String> lines = Files.readAllLines(Paths.get(config.getRuleIdToConfidenceFile().getAbsolutePath()), UTF_8);
+      for (String line : lines) {
+        if (line.startsWith("#")) {
+          continue;
+        }
+        String[] parts = line.split(",");
+        if (parts.length >= 2) {
```

Comment:
Why `>= 2` and not `== 2`?

Reply:
there might be more columns for better debugging, but we don't use them here  - I'll be adding that as a comment